src/auth/auth.controller.ts

```diff
@@ -23,4 +23,18 @@ export class AuthController {
   login(@Body() loginDto: LoginDto): Promise<AuthResponseDto> {
     return this.authService.login(loginDto);
   }
+
+  @Post("reset-password/generate-otp")
+  generateResetPasswordOtp(@Body("email") email: string): Promise<string> {
+    return this.authService.generateResetPasswordOtp(email);
+  }
+
+  @Post("reset-password")
+  resetPassword(
+    @Body("email") email: string,
+    @Body("otp") otpCode: string,
+    @Body("newPassword") newPassword: string,
```

Comment:
Does it make sense to use a dto class and pass the dto itself for the whole Body object?


```suggestion
    @Body() resetPasswordDto: ResetPasswordDto
```